File: app/src/main/java/alejandromr92/com/movies/presentation/model/converters/MovieViewconverter.java
package alejandromr92.com.movies.presentation.model.converters;

import alejandromr92.com.movies.domain.model.Movie;
import alejandromr92.com.movies.presentation.model.MovieView;

import java.util.ArrayList;
import java.util.List;

public abstract class MovieViewconverter {

    public static List<MovieView> convertListToViewModel(List<Movie> domainList){
        List<MovieView> viewList = new ArrayList<>();

        if (domainList != null && domainList.size() > 0){
            for (Movie m: domainList){
                viewList.add(convertToViewModel(m));
            }
        }

        return viewList;
    }

    public static MovieView convertToViewModel(Movie domain){
        MovieView view = new MovieView();

        view.setTitle(domain.getTitle());
        view.setOverview(domain.getOverview());
        view.setPictureUrl(domain.getPictureUrl());
        view.setYear(domain.getYear());

        return view;
    }

    public static List<Movie> convertListToDomainModel(List<MovieView> viewList){
        List<Movie> domainList = new ArrayList<>();

        if (viewList != null && viewList.size() > 0){
            for (MovieView m: viewList){
                domainList.add(convertToDomainModel(m));
            }
        }

        return domainList;
    }

    public static Movie convertToDomainModel(MovieView view){
        Movie domain = new Movie();

        domain.setTitle(view.getTitle());
        domain.setOverview(view.getOverview());
        domain.setPictureUrl(view.getPictureUrl());
        domain.setYear(view.getYear());

        return domain;
    }
}

File: app/src/main/java/alejandromr92/com/movies/presentation/model/converters/MovieViewconverter.kt
package alejandromr92.com.movies.presentation.model.converters

import alejandromr92.com.movies.domain.model.Movie
import alejandromr92.com.movies.presentation.model.MovieView

import java.util.ArrayList

object MovieViewconverter {

    fun convertListToViewModel(domainList: List<Movie>?): List<MovieView> {
        val viewList = ArrayList<MovieView>()

        if (domainList != null && domainList.size > 0) {
            for (m in domainList) {
                viewList.add(convertToViewModel(m))
            }
        }

        return viewList
    }

    fun convertToViewModel(domain: Movie): MovieView {
        val view = MovieView()

        view.title = domain.title
        view.overview = domain.overview
        view.pictureUrl = domain.pictureUrl
        view.year = domain.year

        return view
    }

    fun convertListToDomainModel(viewList: List<MovieView>?): List<Movie> {
        val domainList = ArrayList<Movie>()

        if (viewList != null && viewList.size > 0) {
            for (m in viewList) {
                domainList.add(convertToDomainModel(m))
            }
        }

        return domainList
    }

    fun convertToDomainModel(view: MovieView): Movie {
        val domain = Movie()

        domain.title = view.title
        domain.overview = view.overview
        domain.pictureUrl = view.pictureUrl
        domain.year = view.year

        return domain
    }
}
